libgrust/libproc_macro/ident.h

```diff
@@ -0,0 +1,70 @@
+// Copyright (C) 2023 Free Software Foundation, Inc.
+//
+// This file is part of the GNU Proc Macro Library.  This library is free
+// software; you can redistribute it and/or modify it under the
+// terms of the GNU General Public License as published by the
+// Free Software Foundation; either version 3, or (at your option)
+// any later version.
+
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// Under Section 7 of GPL version 3, you are granted additional
+// permissions described in the GCC Runtime Library Exception, version
+// 3.1, as published by the Free Software Foundation.
+
+// You should have received a copy of the GNU General Public License and
+// a copy of the GCC Runtime Library Exception along with this program;
+// see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see
+// <http://www.gnu.org/licenses/>.
+
+#ifndef IDENT_H
+#define IDENT_H
+
+#include <cstdint>
+#include <string>
+
+struct Ident
+{
+  bool is_raw;
+  unsigned char *val;
```

Comment:
we'll have to handle unicode identifiers as well, so it might be worth putting a note here saying we should take care of this later - this will have repercussions on how we allocate the `val` pointer and what `len` should be - length in bytes or in codepoints, etc

Reply:
The len will always represent the number of bytes. Remember we need to match the rust String type.